src/SimpleGraphs/simpledigraph.jl

```diff
@@ -82,6 +82,149 @@ function SimpleDiGraph(g::AbstractSimpleGraph)
     return h
 end
 
+
+@inbounds function cleanupedges!(fadjlist::Vector{Vector{T}},
+                                 badjlist::Vector{Vector{T}}) where T<:Integer
+    neg = 0
+    for v in 1:length(fadjlist)
+        if !issorted(fadjlist[v])
+            sort!(fadjlist[v])
+        end
+        if !issorted(badjlist[v])
+            sort!(badjlist[v])
+        end
+        unique!(fadjlist[v])
+        unique!(badjlist[v])
+        neg += length(fadjlist[v])
+    end
+    return neg
+end
+
+function SimpleDiGraph(edge_list::Vector{SimpleDiGraphEdge{T}}) where T<:Integer
+    nvg = zero(T)
+    @inbounds(
+    for e in edge_list
```

Comment:
Iterates twice over `edge_list`, this perhaps the right thing to do as it is known to be a vector.

Reply:
I have to do that in order to preallocate the adjacency list as this is much faster.